### PR TITLE
[show] Add subcommand to display the status of auto-restart feature for each container

### PR DESCRIPTION
**Pull request by yozhao101**
**- What I did**
Since we introduced the auto-restart feature for each container, we need add a show subcommand to display the current status of auto-restart feature for all containers or a specific container.

**- How I did it**
We define a function named autorestart to show the status of this features. This function will accept an option parameter which is the container name. If this parameter is not specified, this function will by default show the status of all containers. Otherwise it will show the status of specific container.

**- How to verify it**
We can run the command **show container feature autorestart** or **show container feature autorestart container_name** to verify.

### CODE REVIEW

File: show/main.py [hunk 1]
```diff
@@ -2806,5 +2806,26 @@ def features():
         body.append([key, status_data[key]['status']])
     click.echo(tabulate(body, header))
 
+# show whether the auto-restart feature for containers or a specific container
+# is enabled or disabled.
+
+@cli.command('autorestart')
```
Comment (jleveque) [inline show/main.py]: Since we will have multiple container features including autorestart, this command should be nested in a group (or a couple groups), maybe `show container feature autorestart`. This way we can add other things under both the "container" and "feature" groups in the future.

File: show/main.py [hunk 2]
```diff
@@ -2806,5 +2806,26 @@ def features():
...
+@click.option('-c', '--container-name', required=False)
```
Comment (jleveque) [inline show/main.py]: No need for a flag. You can just add an optional parameter, which is more in line with our other SONiC commands. If the optional parameter is present (e.g., `show container feature autorestart swss`), limit your output to only that container. If it is not present (e.g., `show container feature autorestart`), display the status of all containers. Take a look through this file, there are many examples of this style.

File: show/main.py (hunk 1 above)
Comment (yozhao101) [reply show/main.py]: Please help me review it.

File: show/main.py [hunk 3]
```diff
@@ -2806,5 +2806,41 @@ def features():
         body.append([key, status_data[key]['status']])
     click.echo(tabulate(body, header))
 
+#
+# 'container' group
+#
+@cli.group(cls=AliasedGroup, default_if_no_args=False)
```
Comment (jleveque) [inline show/main.py]: There is no need for this group to be aliased. Try using this definition instead:

```
@cli.group(name='container', invoke_without_command=False)
```

File: show/main.py [hunk 4]
```diff
@@ -2806,5 +2806,41 @@ def features():
         body.append([key, status_data[key]['status']])
     click.echo(tabulate(body, header))
 
+#
+# 'container' group
```
Comment (jleveque) [inline show/main.py]: Provide example after. E.g.,

`# 'container' group (show container ...)`

This helps the reader understand the nesting better.

File: show/main.py [hunk 5]
```diff
@@ -2806,5 +2806,41 @@ def features():
...
+#
+@cli.group(cls=AliasedGroup, default_if_no_args=False)
+def container()
+    """Show container"""
+    pass
+#
+# 'feature' group
+#
+@container.group(cls=AliasedGroup, default_if_no_args=False)
```
Comment (jleveque) [inline show/main.py]: There is no need for this group to be aliased. Try using this definition instead:

```
@container.group(name='feature', invoke_without_command=False)
```

File: show/main.py [hunk 6]
```diff
@@ -2806,5 +2806,41 @@ def features():
...
+    """Show container feature"""
+    pass
+
+# show whether the auto-restart feature for containers or a specific container
+# is enabled or disabled.
+
+@feature.command('autorestart')
+@click.argument('container_name', required=False)
+def autorestart(container_name):
+    """Show container feature autorestart"""
```
Comment (jleveque) [inline show/main.py]: This docstring gets used as the help statement for the command. Please make it a brief explanation of what the command does.

File: show/main.py (hunk 3 above)
Comment (yozhao101) [reply show/main.py]: Fixed.

File: show/main.py (hunk 4 above)
Comment (yozhao101) [reply show/main.py]: Added this example.

File: show/main.py (hunk 5 above)
Comment (yozhao101) [reply show/main.py]: Fixed.

File: show/main.py (hunk 6 above)
Comment (yozhao101) [reply show/main.py]: Added an explanation.

File: show/main.py [hunk 7]
```diff
@@ -2806,5 +2806,44 @@ def features():
...
+
+#
+# 'autorestart' subcommand (show container feature autorestart)
+#
+
+@feature.command('autorestart')
+@click.argument('container_name', required=False)
+def autorestart(container_name):
+    """show whether the auto-restart feature for containers or a specific container
+       is enabled or disabled.
```
Comment (jleveque) [inline show/main.py]: Good, but a bit too wordy. Try, `Show whether the auto-restart feature for container(s) is enabled or disabled`

Also, if the docstring is longer than 45 characters, it will get truncated. However, I discovered recently that one can work around this by defining the parameter `short_help` instead of using a docstring (see https://github.com/pallets/click/issues/486). I think we should probably just move over to this method everywhere. Can you please try that method instead? Example:

```
@feature.command('autorestart', short_help="Show whether the auto-restart feature for container(s) is enabled or disabled")
```

Comment (yozhao101) [reply show/main.py]: Will try the second method.

File: show/main.py [hunk 8]
```diff
@@ -2810,27 +2810,24 @@ def features():
...
+def feature():
     """Show container feature"""
     pass
 
 #
 # 'autorestart' subcommand (show container feature autorestart)
 #
 
-@feature.command('autorestart')
+@feature.command('autorestart', short_help="Show whether the auto-restart feature for contain(s) is enabled or disabled")
```
Comment (jleveque) [inline show/main.py]: Typo: "contain**er**(s):

File: show/main.py (hunk 7 above)
Comment (yozhao101) [reply show/main.py]: I tested the second method on DUT and It did work!

File: show/main.py (hunk 8 above)
Comment (yozhao101) [reply show/main.py]: Fixed.

File: show/main.py [hunk 9]
```diff
@@ -2806,5 +2806,40 @@ def features():
...
+
+#
+# 'autorestart' subcommand (show container feature autorestart)
+#
+@feature.command('autorestart', short_help="Show whether the auto-restart feature for container(s) is enabled or disabled")
+@click.argument('container_name', required=False)
+def autorestart(container_name):
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    header = ['ContainerName', 'Status']
```
Comment (jleveque) [inline show/main.py]: It's OK to add a space in the column header, i.e., `Container Name`

Comment (yozhao101) [reply show/main.py]: Will update.

Comment (yozhao101) [reply show/main.py]: Fixed.
